src/vs/editor/contrib/rtv/RTVSynthDisplay.ts

```diff
@@ -1,113 +1,226 @@
+
+			// Use monaco's monaco-hover class to keep the style the same
+			this.errorHover = document.createElement('div');
+			this.errorHover.className = 'monaco-hover visible';
+			this.errorHover.id = 'snippy-example-hover';
+
+			const scrollable = document.createElement('div');
+			scrollable.className = 'monaco-scrollable-element';
+			scrollable.style.position = 'relative';
+			scrollable.style.overflow = 'hidden';
+
+			const row = document.createElement('row');
+			row.className = 'hover-row markdown-hover';
+
+			const content = document.createElement('div');
+			content.className = 'monaco-hover-content';
+
+			const div = document.createElement('div');
+			const p = document.createElement('p');
+			p.innerText = msg;
+
+			div.appendChild(p);
+			content.appendChild(div);
+			row.appendChild(content);
+			scrollable.appendChild(row);
+			this.errorHover.appendChild(scrollable);
+
+			let position = element.getBoundingClientRect();
+			this.errorHover.style.position = 'fixed';
+			this.errorHover.style.top = position.top.toString() + 'px';
+			this.errorHover.style.left = position.right.toString() + 'px';
+			this.errorHover.style.padding = '3px';
+
+			// Add it to the DOM
+			let editorNode = this.editor.getDomNode()!;
+			editorNode.appendChild(this.errorHover);
+
+			this.errorHover.ontransitionend = () => {
+				if (this.errorHover) {
+					if (this.errorHover.style.opacity === '0') {
+						this.errorHover.remove();
+					}
+				}
+			};
+
+			setTimeout(() => {// TODO Make the error fade over time
+				if (this.errorHover) {
+					this.errorHover.style.transitionDuration = '2s'; // increased from 1s
+					this.errorHover.style.opacity = '0';
+				}
+			}, fadeout);
+		})
+		.catch(err => {
+			if (err) {
+				console.error(err);
+			}
+		});
+	}
+}
+
+export class RTVSynthDisplayBox {
 
-export class RTVSynthDisplayBox implements IRTVDisplayBox{
-	private _box: HTMLDivElement;
-	private _line: HTMLDivElement;
-	private _allEnvs: any[] = [];
-	private _allVars: Set<string> = new Set<string>();
 	private _modeService: IModeService;
 	private _openerService: IOpenerService;
-	// private _deltaVarSet: DeltaVarSet;
-	private _cellDictionary: { [k: string]: [HTMLElement] } = {}; // each HTMLElement is a `td` TableCellElement
+
+	// core elements
+	private _box: HTMLDivElement;
+	private _line: HTMLDivElement;
+	private _errorBox: ErrorHoverManager;
+
+	// from service
+	private _includedTimes: Set<number> = new Set<number>();
+	private _rows: TableElement[][] = [[]];
+	private _boxEnvs: any[] = [];
+
+	// helper data structures/info
+	private _cells?: Map<string, HTMLTableCellElement[]> = undefined;
+	private _currRow: number;
+	private _cursorPos?: CursorPos;
+	private _synthTimer: DelayedRunAtMostOne = new DelayedRunAtMostOne();
+	private _firstEditableCellId?: string = undefined;
+	private _table?: HTMLTableElement;
+	private _cellStyle?: CSSStyleDeclaration;
+
+	exitSynthHandler?: Function;
```

Comment:
Can we switch the `Function` type to the type that includes the parameter(s) and return type?